site/content/entry/citra-progress-report-2017-august.md

```diff
@@ -0,0 +1,176 @@
+with [the path to the virtual SD card](/wiki/user-directory/) added to the beginning.
+
+Now, on top of the usual names for files and folders, there's two special folders
+inside every single folder on your computer, `.` and `..`. These aren't actual
+folders in the sense that you can place files and other folders inside of them.
+Instead, they each symbolize the current folder, and the folder one level above
+it, respectively. For example, `C:/Windows/System32/..` actually means `C:/Windows/`.
+
+With this in mind, a few Citra developers believed a game could, in theory, chain
+multiple `..`s together to get to a file they weren't supposed to know even existed,
+like `/../../../../Documents/IMPORTANT.docx`. Citra would then ask the operating
+system to open the file `%AppData%/Citra/sdmc/../../../../Documents/IMPORTANT.docx`,
+which *actually* means that it would open `C:/Users/Anodium/Documents/IMPORTANT.docx`!
+
+Before you suggest that Citra simply ignore `..`, a game can use it for legitimate
+purposes. And if it were to use it, it would most likely crash, as the resulting
+file path wouldn't exist.
+
+This was already handled for most file functions in Citra, opening, reading,
+writing, etc. except for a few things, like renaming a file. In this case, the
+malicious game could just ask Citra to change `/../../../../Documents/IMPORTANT.docx`'s
+name to `/delicious_secrets.docx`, effectively moving `IMPORTANT.docx` into Citra's
+virtual SD card! From there, the game could just ask Citra to open
+`/delicious_secrets.docx` to read the file it was originally forbidden from accessing.
+
+This patch now fixes this, such that if any 3DS game or application tries to do
+exploit the rename file function from inside Citra, rather than doing what it asks,
+Citra gives them `ERROR_INVALID_PATH`, which most games interpret by crashing. So
+far, we haven't found any real 3DS software that tries to do this, but at least
+now future attempts to do so are blocked.
+
+Do note though, that most of us are not security experts, and even for those who are,
+harderning the software is much more difficult and much less worthwhile than actually
+having an accurate 3DS emulation. The surface area of an emulation project is huge,
+and so writing malicious code intending to exploit one is not too difficult in any case.
+
+But at the same time, malware authors generally target either the largest populations,
+or the most lucrative populations, and emulation is neither. And for the effort
+required to do so, most will only toy with the idea in their head, or maybe write
+a proof-of-concept for fun, rather genuine malicious intent.
+
+Regardless, this demonstrates the second reason we are against piracy and unofficial
+Citra builds. That with enough effort, it is very possible to spread malware through
+either. (The first is that it's simply a blatant violation of copyright laws.)
```

Comment:
This paragraph doesn't read well at all - revise? Don't think there is any value in this as it stands.